src/federation_scheduled_tasks.rs

```diff
@@ -0,0 +1,106 @@
+use chrono::Duration;
+use clokwerk::{AsyncScheduler, TimeUnits};
+use diesel::{Connection, PgConnection, QueryDsl, RunQueryDsl};
+use lemmy_api_common::context::LemmyContext;
+use lemmy_apub::DEAD_INSTANCES;
+use lemmy_db_schema::{
+  schema::instance,
+  source::instance::{Instance, InstanceForm},
+  utils::naive_now,
+};
+use lemmy_routes::nodeinfo::NodeInfo;
+use lemmy_utils::error::{LemmyError, LemmyResult};
+use tokio::sync::OnceCell;
+use tracing::{error, info};
+
+pub async fn setup_federation_scheduled_tasks(
+  db_url: String,
+  context: LemmyContext,
+) -> Result<(), LemmyError> {
+  let mut scheduler = AsyncScheduler::new();
+
+  // Check for dead federated instances
+  static CONTEXT: OnceCell<LemmyContext> = OnceCell::const_new();
```

Comment:
Shouldn't the DeadInstances be the only thing you need in a OnceCell? Why are these other ones needed.